doc: clarify BOARD_ROOT documentation

hallo-alex opened the pull request:
clarify BOARD_ROOT must point to dir ABOVE boards

hallo-alex committed:
Update index.rst

clarify BOARD_ROOT must point to dir ABOVE boards

ulfalizer commented on doc/application/index.rst:
```diff
@@ -580,7 +580,7 @@ targeting this board by specifying the location of your custom board
 build system::
 
-   cmake -DBOARD=<board name> -DBOARD_ROOT=<path to boards> ..
+   cmake -DBOARD=<board name> -DBOARD_ROOT=<path the directory containig a "boards" directory> ..
```
Language nit: "path to directory containing a boards/ directory"

dbkinder replied:
... also fixes the misspelling of "containing"  :)

dbkinder reviewed (changes requested):
There are lots of index.rst files in the documentation. The PR title should be more informative, something like:
```
doc: clarify BOARD_ROOT documentation
```

hallo-alex committed:
doc: clarify BOARD_ROOT documentation

suggestions from https://github.com/zephyrproject-rtos/zephyr/pull/14021#discussion_r261884311 and https://github.com/zephyrproject-rtos/zephyr/pull/14021#pullrequestreview-210332420

aescolar commented:
Please squash both commits together. In general, when fixing a pull request commit, instead of creating a new one, amend the previous one.
Also please address the issue reported by gitlint (you need to add a signed off message).

aescolar commented:
@hallo-alex : Please note that until the issues reported in https://github.com/zephyrproject-rtos/zephyr/pull/14021#issuecomment-469057614 are not fixed, and the commits are not squashed this commit PR cannot be accepted.

aescolar reviewed (changes requested):
Please:
* squash both commits together
* Add a signed off line to the commit message (you can use "git commit --amend -s")
